lib/teleterm/api/proto/v1/tshd_events_service.proto

```diff
@@ -19,17 +19,57 @@ package teleport.terminal.v1;
 option go_package = "github.com/gravitational/teleport/lib/teleterm/v1";
 
 // TshdEventsService is served by the Electron app. The tsh daemon calls this service to notify the
-// app about actions that happen outside of the app itself. For example, when the user tries to
-// connect to a gateway served by the daemon but the cert has since expired and needs to be
-// reissued.
+// app about actions that happen outside of the app itself.
 service TshdEventsService {
-  // Test is an RPC that's used to demonstrate how the implementation of a tshd event may look like
-  // from the beginning till the end.
-  // TODO(ravicious): Remove this once we add an actual RPC to tshd events service.
-  rpc Test(TestRequest) returns (TestResponse);
+  // Relogin makes the Electron app display a login modal for the specific root cluster. The request
+  // returns a response after the relogin procedure has been successfully finished.
+  rpc Relogin(ReloginRequest) returns (ReloginResponse);
+  // SendNotification causes the Electron app to display a notification in the UI. The request
+  // accepts a specific message rather than a generic string so that the Electron is in control as
+  // to what message is displayed and how exactly it looks.
+  rpc SendNotification(SendNotificationRequest) returns (SendNotificationResponse);
 }
 
-message TestRequest {
-  string foo = 1;
+// Relogin
+
+message ReloginRequest {
+  string root_cluster_uri = 1;
+  oneof reason {
+    GatewayCertExpired gateway_cert_expired = 2;
+  }
+}
+
+// GatewayCertExpired is given as the reason when something attempts to make a connection through
+// the gateway but:
+//
+// * the db cert used by the gateway has expired
```

Comment:
> A bit unclear, looks like it says that we do this in 2 cases -
> 
>     1. when db cert expired on gateway,
> 
>     2. when both db and user cert expired.
> 
> 
> But as I understand it doesn't happen for the 1?

Here I wanted to list all conditions that must be met for `GatewayCertExpired` to be used as the reason but I see how it might give an impression that it can happen in two separate cases. I'll remove the list from here and just write it in a sentence.